src/Microsoft.AspNet.Hosting/Internal/HostingEngine.cs

```diff
@@ -87,6 +88,7 @@ public virtual IDisposable Start()
                 _applicationLifetime.NotifyStopping();
                 server.Dispose();
                 _applicationLifetime.NotifyStopped();
+                loggerFactory.Dispose();
```

Comment:
Hmm...I have seen DI scopes used in `per-request` kind of scenarios where as logger factory here is an application wide service. (thinking of it in a different way...the application start and shutdown itself is a scope and probably we should just dispose off all disposable application services as how DI scope would do).